org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/markers/SonarMarker.java

```diff
@@ -97,76 +95,44 @@ private static Integer getPriority(final String severity) {
     return result;
   }
 
-  public static void addLine(final Map<String, Object> markerAttributes, final long line, final IResource resource) {
+  // Visible for testing
+  public static void setPosition(final Map<String, Object> markerAttributes, final Issue issue, final IResource resource) {
     if (resource instanceof IFile) {
-      IFile file = (IFile) resource;
-      try (InputStream is = file.getContents(); LineAndCharCountReader lnr = new LineAndCharCountReader(new InputStreamReader(is, file.getCharset()))) {
-        while (lnr.read() != -1) {
-          if (lnr.getLineNumber() == line) {
-            markerAttributes.put(IMarker.CHAR_START, lnr.getCharIndex());
-            String s = lnr.readLine();
-            if (s != null) {
-              markerAttributes.put(IMarker.CHAR_END, lnr.getCharIndex() + s.length());
-            }
-            return;
-          }
+      IFile iFile = (IFile) resource;
+      Integer startLine = issue.getStartLine();
+      if (startLine == null) {
+        return;
+      }
+      ITextFileBufferManager iTextFileBufferManager = FileBuffers.getTextFileBufferManager();
+      try {
+        iTextFileBufferManager.connect(iFile.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
+        ITextFileBuffer iTextFileBuffer = iTextFileBufferManager.getTextFileBuffer(iFile.getFullPath(), LocationKind.IFILE);
+        IDocument iDoc = iTextFileBuffer.getDocument();
+        int startLineStartOffset = iDoc.getLineOffset(startLine - 1);
+        Integer issueStartLineOffset = issue.getStartLineOffset();
+        if (issueStartLineOffset != null) {
+          markerAttributes.put(IMarker.CHAR_START, startLineStartOffset + issueStartLineOffset);
+          Integer issueEndLine = issue.getEndLine();
+          int endLineStartOffset = issueEndLine != startLine ? iDoc.getLineOffset(issueEndLine - 1) : startLineStartOffset;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)